pkg/ec2pricing/ec2pricing.go

```diff
@@ -0,0 +1,110 @@
+package ec2pricing
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/pricing"
+	"github.com/aws/aws-sdk-go/service/pricing/pricingiface"
+)
+
+// EC2Pricing is the public struct to interface with AWS pricing APIs
+type EC2Pricing struct {
+	PricingClient pricingiface.PricingAPI
+	AWSSession    *session.Session
+}
+
+// EC2PricingIface is the EC2Pricing interface mainly used to mock out ec2pricing during testing
+type EC2PricingIface interface {
+	GetOndemandInstanceTypeCost(instanceType string) (float64, error)
+}
+
+// New creates an instance of instance-selector EC2Pricing
+func New(sess *session.Session) EC2Pricing {
+	return EC2Pricing{
+		PricingClient: pricing.New(sess),
+		AWSSession:    sess,
+	}
+}
+
+// GetOndemandInstanceTypeCost retrieves the on-demand hourly cost for the specified instance type
+func (p EC2Pricing) GetOndemandInstanceTypeCost(instanceType string) (float64, error) {
+	endpointResolver := endpoints.DefaultResolver()
+	partitions := endpointResolver.(endpoints.EnumPartitions).Partitions()
+	// use us-east-1 as the default
+	regionDescription := "US East (N. Virginia)"
```

Comment:
why not use the user's region settings for pricing?

Reply:
this does use the user's region. I set a default just in case the region description changes in the SDK. The pricing API only uses region description rather than the region code. And there's a comment in the `endpoints` pkg in the sdk that the description could change. So to be extra cautious I set a default that is used if no region match can be found.